report_builder/models.py

```diff
@@ -85,18 +88,18 @@ def save(self, *args, **kwargs):
             unique_slugify(self, self.name)
         super(Report, self).save(*args, **kwargs)
 
-    def add_aggregates(self, queryset):
-        for display_field in self.displayfield_set.filter(aggregate__isnull=False):
-            if display_field.aggregate == "Avg":
-                queryset = queryset.annotate(Avg(display_field.path + display_field.field))
-            elif display_field.aggregate == "Max":
-                queryset = queryset.annotate(Max(display_field.path + display_field.field))
-            elif display_field.aggregate == "Min":
-                queryset = queryset.annotate(Min(display_field.path + display_field.field))
-            elif display_field.aggregate == "Count":
-                queryset = queryset.annotate(Count(display_field.path + display_field.field))
-            elif display_field.aggregate == "Sum":
-                queryset = queryset.annotate(Sum(display_field.path + display_field.field))
+    def add_aggregates(self, queryset, display_fields=None):
```

Comment:
I did review your work @grantjenks and this should look very familiar - thanks for that change.

It may be easier to read without seeing the diff.
https://github.com/burke-software/django-report-builder/blob/master/report_builder/models.py#L91

It doesn't affect self. It uses self only to gather display fields if they are not explicitly passed. Saul if this isn't clear let me know.